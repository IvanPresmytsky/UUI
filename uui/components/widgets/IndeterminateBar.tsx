import * as React from 'react';
import cx from 'classnames';
import { IHasCX } from '@epam/uui-core';
import css from './IndeterminateBar.module.scss';

export interface IndeterminateBarMods extends IHasCX, React.RefAttributes<HTMLDivElement> {
    /** Component size */
    size?: '12' | '18' | '24';
}

export const IndeterminateBar = (props: IndeterminateBarMods) => {
    return (
        <div ref={ props.ref } className={ cx('uui-indeterminate_bar', props.cx, css.root, css[`size-${props.size || 12}`]) }>
            <div className={ cx(css.progressBar) } />
        </div>
    );
};
