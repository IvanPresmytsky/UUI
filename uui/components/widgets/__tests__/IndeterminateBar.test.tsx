import React from 'react';
import { IndeterminateBar } from '../IndeterminateBar';
import { renderer } from '@epam/uui-test-utils';

describe('ProgressBar', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<IndeterminateBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer.create(<IndeterminateBar size="24" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
