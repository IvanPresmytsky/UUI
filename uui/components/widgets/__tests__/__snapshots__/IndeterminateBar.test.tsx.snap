// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should be rendered correctly 1`] = `
<div
  className="uui-indeterminate_bar root size-12"
>
  <div
    className="progressBar"
  />
</div>
`;

exports[`ProgressBar should be rendered correctly with props 1`] = `
<div
  className="uui-indeterminate_bar root size-24"
>
  <div
    className="progressBar"
  />
</div>
`;
